Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -116,271 +51,93 @@ COREGRAPHICS_EXPORT void CGImageAddDestructionListener(CGImageDestructionListene
-        image cropped to the specified rectangle.
+ @Notes Limited bitmap formats available and decode is not supported.
 */
-CGImageRef CGImageCreateWithImageInRect(CGImageRef ref, CGRect rect) {
-    if (ref == NULL) {
-        TraceWarning(TAG, L"CGImageCreateWithImageInRect: ref = NULL!");
-        return 0;
-    }
-
-    rect = CGRectIntegral(rect);
-
-    CGRect imgRefSize;
-    imgRefSize.origin.x = 0;
-    imgRefSize.origin.y = 0;
-    imgRefSize.size.width = (float)ref->Backing()->Width();
-    imgRefSize.size.height = (float)ref->Backing()->Height();
-
-    rect = CGRectIntersection(rect, imgRefSize);
-
-    __CGSurfaceInfo surfaceInfo;
-    ref->Backing()->GetSurfaceInfoWithoutPixelPtr(&surfaceInfo);
-
-    // Override width and height with the rect
-    surfaceInfo.width = rect.size.width;
-    surfaceInfo.height = rect.size.height;
-
-    assert(surfaceInfo.surfaceData == NULL);
+CGImageRef CGImageCreate(size_t width,
+                         size_t height,
+                         size_t bitsPerComponent,
+                         size_t bitsPerPixel,
+                         size_t bytesPerRow,
+                         CGColorSpaceRef colorSpace,
+                         CGBitmapInfo bitmapInfo,
+                         CGDataProviderRef provider,
+                         const float* decode,
+                         bool shouldInterpolate,
+                         CGColorRenderingIntent intent) {
+    RETURN_NULL_IF(((provider == nullptr) || ![(NSObject*)provider isKindOfClass:[NSData class]]) || (colorSpace == nullptr));
 
-    CGImageRef newImage = new CGBitmapImage(surfaceInfo);
+    NSData* dataProvider = (__bridge NSData*)provider;
 
-    int startX = (int)rect.origin.x;
-    int startY = (int)rect.origin.y;
-    int sizeX = (int)rect.size.width;
-    int sizeY = (int)rect.size.height;
+    unsigned char* data = (unsigned char*)[dataProvider bytes];
 
-    BYTE* srcIn =
-        ((BYTE*)ref->Backing()->LockImageData()) + startY * ref->Backing()->BytesPerRow() + startX * ref->Backing()->BytesPerPixel();
-    BYTE* destOut = (BYTE*)newImage->Backing()->LockImageData();
+    ComPtr<IWICBitmap> image;
+    ComPtr<IWICImagingFactory> imageFactory = _GetWICFactory();
 
-    for (int curY = 0; curY < sizeY; curY++) {
-        memmove(destOut, srcIn, newImage->Backing()->BytesPerRow());
+    REFGUID pixelFormat = _CGImageGetWICPixelFormat(bitsPerComponent, bitsPerPixel, colorSpace, bitmapInfo);
 
-        srcIn += ref->Backing()->BytesPerRow();
-        destOut += newImage->Backing()->BytesPerRow();
-    }
+    HRESULT status = imageFactory->CreateBitmapFromMemory(width, height, pixelFormat, bytesPerRow, height * bytesPerRow, data, &image);
 
-    ref->Backing()->ReleaseImageData();
-    newImage->Backing()->ReleaseImageData();
+    CGImageRef imageRef = __CGImage::CreateInstance();
+    imageRef->SetImageSource(image).SetColorSpace(colorSpace).SetRenderingIntent(intent).SetInterpolate(shouldInterpolate);
```

Comment:
>SetImageSource(image).SetColorSpace(colorSpace).SetRenderingIntent(intent).SetInterpolate(shouldInterpolate); [](start = 14, length = 109)

This pattern is cute but I'd much rather have these all as separate statements or better yet add these as arguments to the CreateInstance call as you are basically doing a .ctor like operation here ...